freqtrade/optimize/__init__.py

```diff
@@ -1,41 +1,112 @@
 # pragma pylint: disable=missing-docstring
 
-
+import logging
 import json
 import os
 from typing import Optional, List, Dict
+import time
+from freqtrade.exchange import get_ticker_history
 
 from pandas import DataFrame
 
 from freqtrade.analyze import populate_indicators, parse_ticker_dataframe
 
+logger = logging.getLogger(__name__)
 
-def load_data(ticker_interval: int = 5, pairs: Optional[List[str]] = None) -> Dict[str, List]:
+def load_data(pairs: List[str], ticker_interval: int = 5, refresh_pairs: Optional[bool] = False) -> Dict[str, List]:
     """
     Loads ticker history data for the given parameters
     :param ticker_interval: ticker interval in minutes
     :param pairs: list of pairs
     :return: dict
     """
-    path = os.path.abspath(os.path.dirname(__file__))
+    path = testdata_path()
     result = {}
-    _pairs = pairs or [
-        "BTC_ETH", "BTC_LTC", "BTC_ETC", "BTC_DASH", "BTC_ZEC",
-        "BTC_XLM", "BTC_NXT", "BTC_POWR", "BTC_ADA", "BTC_XMR",
-    ]
-    for pair in _pairs:
-        with open('{abspath}/../tests/testdata/{pair}-{ticker_interval}.json'.format(
+
+    # If the user force the refresh of pairs
+    if refresh_pairs:
+        logger.info('Download data for all pairs and store them in freqtrade/tests/testsdata')
+        download_pairs(pairs)
+
+    for pair in pairs:
+        file = '{abspath}/{pair}-{ticker_interval}.json'.format(
             abspath=path,
             pair=pair,
             ticker_interval=ticker_interval,
-        )) as tickerdata:
+        )
+        # The file does not exist we download it
+        if not os.path.isfile(file):
+            download_backtesting_testdata(pair=pair, interval=ticker_interval)
+
+        # Read the file, load the json
+        with open(file) as tickerdata:
             result[pair] = json.load(tickerdata)
     return result
 
-
 def preprocess(tickerdata: Dict[str, List]) -> Dict[str, DataFrame]:
     """Creates a dataframe and populates indicators for given ticker data"""
     processed = {}
     for pair, pair_data in tickerdata.items():
         processed[pair] = populate_indicators(parse_ticker_dataframe(pair_data))
     return processed
+
+def testdata_path() -> str:
+    """Return the path where testdata files are stored"""
+    return os.path.abspath(os.path.dirname(__file__)) + '/../tests/testdata'
+
+def download_pairs(pairs: List[str]) -> bool:
+    """For each pairs passed in parameters, download 1 and 5 ticker intervals"""
+    for pair in pairs:
+        try:
+            for interval in [1,5]:
+                download_backtesting_testdata(pair, interval=interval)
+        except BaseException:
+            time.sleep(40)
```

Comment:
This sleep can also be removed (see other comment few lines below). I think `logger.exception()` or `logger.warning()` would be useful to give some indication what went wrong